src/gfw/common/bigquery/table_description.py

```diff
@@ -0,0 +1,105 @@
+"""Provides a class for generating structured BigQuery table descriptions.
+
+This module defines the `TableDescription` dataclass, which produces
+a standardized, human-readable description string for use in
+BigQuery table metadata. The description includes a title, subtitle,
+summary, caveats, and a formatted list of relevant parameters.
+
+The formatting is designed to be readable in the BigQuery UI,
+GitHub, and other plaintext contexts.
+"""
+
+from dataclasses import dataclass, field
+from typing import Any
+
+
+DESCRIPTION_TEMPLATE = """\
+❖ {title} ❖
+{subtitle}.
+⬖ Created by {project}: v{version}.
+⬖ https://github.com/GlobalFishingWatch/{project}.
+
+𝗦𝘂𝗺𝗺𝗮𝗿𝘆
+{summary}
+
+𝗖𝗮𝘃𝗲𝗮𝘁𝘀
+{caveats}
+
+For more information, see https://github.com/GlobalFishingWatch/{project}/blob/HEAD/README.md.
+
+𝗥𝗲𝗹𝗲𝘃𝗮𝗻𝘁 𝗽𝗮𝗿𝗮𝗺𝗲𝘁𝗲𝗿𝘀
+{relevant_params}
+"""  # noqa
+
+TO_BE_COMPLETED = "To be completed."
+
+
+@dataclass
+class TableDescription:
+    """Generates a structured description for BigQuery table metadata.
+
+    Attributes:
+        project:
+            GitHub project or repo name (used for URLs and headers).
+
+        version:
+            Version of the project or schema.
+
+        title:
+            Title of the table or dataset.
+
+        subtitle:
+            Subtitle or one-line summary.
+
+        summary:
+            High-level summary of the table's purpose.
+
+        caveats:
+            Known limitations or notes about the data.
+
+        relevant_params:
+            A dictionary of key parameters relevant to the table's content or generation.
+            The keys are parameter names (strings), and the values can be any type convertible
+            to string.
+
+            When rendered, the parameters are formatted as aligned key-value pairs with colons
+            aligned in a vertical column, for example:
+
+                param1      : value1
+                long_param2 : value2
+                x           : 42
+    """
+
+    project: str
```

Comment:
What do you think of using `repo_name` for this variable so that it is not misunderstood with the BigQuery project?

Reply:
Done.